### ISSUE TITLE
Upgrade wiki.openstreetmap.org to mediawiki 1.39 

### ISSUE THREAD

**Issue by Firefishy** [openstreetmap/operations#760]
Upgrade wiki.openstreetmap.org to Mediawiki 1.39 once released.

Requirements: PHP8 which is default in Ubuntu 22.04.

Some servers have not yet been upgraded to Ubuntu 22.04 because 22.04 installs PHP8 which is not compatible with Mediawiki <1.39.

**Comment by Firefishy** [comment]
Linked Issue: https://github.com/openstreetmap/operations/issues/749

**Comment by Firefishy** [comment]
Linked Issue: https://github.com/openstreetmap/operations/issues/659 (22.04 upgrades)

**Comment by Firefishy** [comment]
Linked Issue: https://github.com/openstreetmap/operations/issues/735 (Low disk space on tabaluga aka wiki.osm.org)

**Comment by tomhughes** [comment]
I upgraded everything except the main wiki to 1.37 some months ago but then never got around to doing the main wiki and 1.38 is out now so I upgraded them to that last week with the intention of doing the main wiki if no problems showed up.

Doing the main wiki is a much bigger risk though, mostly due to wikidata stuff, as even mediawiki people often seem unsure what versions of that will work or how to configure them - probably because they're all insane and are running bleeding edge 1.40 code on their production sites for both wikipedia and wikidata!

Obviously we do also need a solution (yet again) to map extensions that get abandoned as soon as they're written.

**Comment by Firefishy** [comment]
Amended ticket to reflect that only wiki.osm.org needs upgrade.

**Comment by tomhughes** [comment]
Well the others will need to go 1.39 as well so we can upgrade ridley to 22.04 but that's a relatively easy job normally once it is released.

**Comment by GreenReaper** [comment]
FWIW you shouldn't *need* to upgrade to PHP 8.0 just because that is what Ubuntu 22.04 ships with. You can install [ppa:ondrej/php](https://launchpad.net/~ondrej/+archive/ubuntu/php) as a software repo and put -7.4 packages on (or whatever).

Conversely, if you *do* want to upgrade to PHP 8.0, 
the same PPA should work to enable versions of PHP 8.x on older versions, in parallel with the existing one. [Recent release branches of MediaWiki include support](https://phabricator.wikimedia.org/T248925#8249837) and AFAIK changes were merged into at least MW 1.35 and possibly 1.37, as well as the 1.38 and 1.39 mentioned there. Some extensions may be another matter, but usually they Wikimedia support the ones they use.

**Comment by Firefishy** [comment]
> FWIW you shouldn't _need_ to upgrade to PHP 8.0 just because that is what Ubuntu 22.04 ships with. You can install [ppa:ondrej/php](https://launchpad.net/~ondrej/+archive/ubuntu/php) as a software repo and put -7.4 packages on (or whatever).
> 

Thanks. We try only run support distro software releases and keep PPA usage to a minimal. The larger the fleet and the more bespoke each system setup is the more difficult it is to maintain. PPA has no guarantees about security updates etc.

**Comment by GreenReaper** [comment]
That's understandable. I would note that the maintainer of the PPA is also [the maintainer of oHP packaging for Debian](https://deb.sury.org/#donate), on which Ubuntu's PHP packages are based, and is [a member of Ubuntu's PHP packaging team](https://launchpad.net/~php-ubuntu)..

Overall I'd still recommend going to 8.x, just wanted you to be aware it is an option not to (or to test on an older server in parallel). Wikimedia themselves stuck with 7.2 for the longest time, and [only relatively recently went to 7.4](https://www.wikidata.org/wiki/Special:Version).

**Comment by tomhughes** [comment]
Upgrade of the main wiki is now complete so we should be good to retry the move of the wiki to konqui that is running Ubuntu 22.04 and PHP 8.
